File: Advanced Methods of Programming/Toy Language Interpreter/src/Model/Expressions/Variable.java
package Model.Expressions;

import Model.ProgramState;

public class Variable implements IExpression {

    private String id;

    public Variable(String id){
        this.id = id;
    }

    @Override
    public int eval(ProgramState state) {
        return state.getSymTable().get(id);
    }

    @Override
    public String toString() {
        return id;
    }
}

File: Advanced Methods of Programming/CSharp Toy Language Interpreter/ToyLanguageInterpreter/Src/Model/Expressions/Variable.cs
using System;
using ToyLanguageInterpreter.Src.Model;
using ToyLanguageInterpreter.Src.Model.Expressions;

namespace ToyLanguageInterpreter.Src.View
{
    internal class Variable : IExpression
    {
        private string id;

        public Variable(string id)
        {
            this.id = id;
        }

        public int Eval(ProgramState state)
        {
            return state.SymTable.Get(id);
        }

        public override string ToString()
        {
            return id;
        }
    }
}